src/script/components/MessagesList/UploadAssets/components/UploadAssetItem/UploadAssetItem.tsx

```diff
@@ -0,0 +1,56 @@
+/*
+ * Wire
+ * Copyright (C) 2024 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {useEffect, useState} from 'react';
+
+import {GenericMessage} from '@wireapp/protocol-messaging';
+
+import {LoadingBar} from 'Components/LoadingBar';
+
+import {uploadingProgressText} from './UploadAssetItem.styles';
+
+import {AssetRepository} from '../../../../../assets/AssetRepository';
+
+interface Props {
+  assetRepository: AssetRepository;
+  message: GenericMessage;
+}
+
+export const UploadAssetItem = ({assetRepository, message}: Props) => {
+  const [uploadProgress, setUploadProgress] = useState<number>(0);
```

Comment:
minor, the `number` here seems unnecessary, TypeScript is going to infer the type: 
```suggestion
  const [uploadProgress, setUploadProgress] = useState(0);
```

Reply:
I actually like the explicit approach, this way later on if you try to do setUploadProgress("somestring") typescript will have an error that this value must be a number

Reply:
If the initial value is `0`, then TS won't let you pass anything else than a number:

<img width="408" alt="Screenshot 2024-11-26 at 10 08 42" src="https://github.com/user-attachments/assets/4a164988-b9ae-47e4-820c-45d199ecaa92">

Reply:
Yee, if You add default value it's automatically getting type ;)